File: 1036-rotting-oranges/rotting-oranges.java
class Solution {
    public int orangesRotting(int[][] grid) {
        boolean oneexists = false;
        Queue<int[]> qu = new LinkedList<>();
        for(int i = 0; i < grid.length; i++){
            for(int j = 0; j < grid[0].length; j++){
                if(grid[i][j] == 2){
                    qu.add(new int[] {i, j});
                }
                if(grid[i][j] == 1){
                    oneexists = true;
                }
            }
        }
        if(!oneexists){
            return 0;
        }
        int level = 0;
        while(!qu.isEmpty()){
            int size = qu.size();
            while(size > 0){
                int[] curr = qu.poll();
                int x = curr[0];
                int y = curr[1];
                if(x < grid.length - 1 && grid[x + 1][y] == 1){
                    grid[x+1][y] = 2;
                    qu.add(new int[] {x+1, y});
                }
                if(y < grid[0].length - 1 && grid[x][y + 1] == 1){
                    grid[x][y + 1] = 2;
                    qu.add(new int[] {x, y+1});
                }                
                if(x > 0 && grid[x - 1][y] == 1){
                    grid[x-1][y] = 2;
                    qu.add(new int[] {x-1, y});
                }                
                if(y > 0 &&  grid[x][y - 1] == 1){
                    grid[x][y-1] = 2;
                    qu.add(new int[] {x, y-1});
                }
                size--;
            }
            level++;
        }
        boolean flag = false;
        for(int x = 0; x < grid.length; x++){
            for(int y = 0; y < grid[0].length; y++){
                if(grid[x][y] == 1){
                    flag = true;
                }
            }
        }
        if(flag){
            return -1;
        }
        return level - 1;
    }
}

File: 1036-rotting-oranges/rotting-oranges.py
class Solution:
    def orangesRotting(self, grid: List[List[int]]) -> int:
        qu = deque()
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 2:
                    qu.append((i,j))
        level = 0
        result = 0
        while len(qu) > 0:
            size = len(qu)
            while size > 0:
                cell = qu.popleft()
                i = cell[0]
                j = cell[1]
                if i > 0:
                    if grid[i-1][j] == 1:
                        grid[i-1][j] = -2
                        result = max(result, level + 1)
                        qu.append((i-1,j))
                if j > 0:
                    if grid[i][j - 1] == 1:
                        grid[i][j - 1] = -2
                        result = max(result, level + 1)
                        qu.append((i,j - 1))
                if i < len(grid) - 1:
                    if grid[i+1][j] == 1:
                        grid[i+1][j] = -2
                        result = max(result, level + 1)
                        qu.append((i+1,j))
                if j < len(grid[0]) - 1:
                    if grid[i][j + 1] == 1:
                        grid[i][j + 1] = -2
                        result = max(result, level + 1)
                        qu.append((i,j + 1))
                size = size - 1
            level = level + 1
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    return -1
        return result


         